### ISSUE TITLE
How to distribute all the @Test methods from all classes across all connected devices in parallel ?

### ISSUE THREAD

**Issue by VinitSandesara** [saikrishna321/AppiumTestDistribution#4]
I tried to execute @test method thru "parallel" and "distribute" both works the way it was implemented however for distribute instead of distributing per class can we distribute per @test method

Lets say i have 2 class each are having total 6 @test method ( class 1 with 6 @test method and class 2 with 6 @test method) so total 12 @test method are available to execute on lets say 3 connected android devices but i wanna execute all these 12 @test method parallel means if the 1st device took one @test method it should not execute on other connected devices, as soon as device finished executing @test method it should start execute one on pipeline (waiting) and this way all 3 devices continue executing until there is no @test method left to be execute in pipeline (waiting).

Long story short what i want is all the @Test methods from all classes to get distributed across connected devices ?

Note : the project i work on we have this requirement to follow, if you can please help Sai, i really appreciated if there is any workaround you can provide me for above.

Thanks
Vinit

**Comment by VinitSandesara** [comment]
Hi Sai, Any updates on this enhancement ? like what is an ETA we should expect.

Thanks
Vinit

**Comment by saikrishna321** [comment]
There is bug on java_client Appium end, Appium server deadlocks when we trigger on method level.

 I had raised an issue seems its fixed but I can still reproduce it. 

Will be able to get it implemented once that issue is fixed

**Comment by VinitSandesara** [comment]
Hi Sai, Any updates on this ?  any ETA from java_client side to fix the bug and release it out ?

Thanks
Vinit

**Comment by VinitSandesara** [comment]
Hi Sai, Any updates on this ? 

Thanks
Vinit

**Comment by saikrishna321** [comment]
Is implemented as part of 7.0.1
